File: Java OOP-Feb2020/src/j_polymorphism_Exercises/wildFarm/Mammal.java
package j_polymorphism_Exercises.wildFarm;

import java.text.DecimalFormat;

public abstract class Mammal extends Animal {
    private String livingRegion;

    public Mammal(String animalType, String animalName, Double animalWeight, String livingRegion) {
        super(animalType, animalName, animalWeight);
        this.livingRegion = livingRegion;
    }

    protected String getLivingRegion() {
        return livingRegion;
    }

    @Override
    public String toString() {
        return String.format("%s[%s, %s%s, %s, %d]",
                this.getAnimalType(),
                this.getAnimalName(),
                this.addSpecialText(),
                new DecimalFormat("#.##").format(this.getAnimalWeight()),
                this.getLivingRegion(),
                this.getFoodEaten());
    }

    protected String addSpecialText() {
        return "";
    }
}

File: Programming-for-QA-sep2023/21.Implementing-OOP-Hierarchy/WildFarm/Animal/Mammal.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace WildFarm.Animal;

public abstract class Mammal : Animal
{
    protected Mammal(string name, double weight, string livingRegion) : base(name, weight)
    {
        this.LivingRegion = livingRegion;
    }

    public string LivingRegion { get; set; }

    public override string ToString()
    {
        return $"{this.GetType().Name} [{this.Name}, {this.Weight}, {this.LivingRegion}, {this.FoodEaten}]";
    }
}
